modules/dockermodelrunner/internal/sdk/client/pull.go

```diff
@@ -18,19 +20,48 @@ func (c *Client) PullModel(ctx context.Context, fullyQualifiedModelName string)
 		return fmt.Errorf("new post request (%s): %w", reqURL, err)
 	}
 
-	log.Default().Printf("🙏 Pulling model %s. Please be patient, no progress bar yet!", fullyQualifiedModelName)
+	log.Default().Printf("🙏 Pulling model %s. Please be patient", fullyQualifiedModelName)
+
+	// Check context before making request
+	if err := ctx.Err(); err != nil {
+		return fmt.Errorf("context done before request: %w", err)
+	}
 
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
 		return fmt.Errorf("http post: %w", err)
 	}
 	defer resp.Body.Close()
 
+	// Check context after getting response
+	if err := ctx.Err(); err != nil {
+		return fmt.Errorf("context done after response: %w", err)
+	}
+
 	// The Docker Model Runner returns a 200 status code for a successful pulls
 	if resp.StatusCode != http.StatusOK {
 		return fmt.Errorf("unexpected status code: %d", resp.StatusCode)
 	}
 
+	scanner := bufio.NewScanner(resp.Body)
+	done := make(chan error, 1)
+
+	go func() {
+		for scanner.Scan() {
+			log.Default().Printf(scanner.Text())
+		}
+		done <- scanner.Err()
+	}()
+
+	select {
+	case <-ctx.Done():
+		return fmt.Errorf("context done: %w", ctx.Err())
+	case err := <-done:
+		if err != nil {
+			return fmt.Errorf("scan error: %w", err)
+		}
+	}
```

Comment:
suggestion: no need for the go routine here as scanner will return the context error if needed from the its `Read` call to `resp.Body`.
```suggestion
	scanner := bufio.NewScanner(resp.Body)

	// TODO: use a progressbar instead of multiple line output.
	logger := log.Default()
	for scanner.Scan() {
		logger.Printf(scanner.Text())
	}
	
	if err := scanner.Err(); err != nil {
		return fmt.Errorf("scan error: %w", err)
	}
```